components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -983,11 +989,57 @@ public Response apisApiIdSubscriptionPoliciesGet(String apiId, String xWSO2Tenan
         return null;
     }
 
+    private void addLatestMajorRangeVersionDetails(APIConsumer apiConsumer, ApiTypeWrapper api, String organization)
+            throws APIManagementException {
+        if (!api.getVersion().startsWith("v")) {
+            return;
+        }
+        String apiProvider = api.getId().getProviderName();
+        SemVersion semVersion = SemanticVersionUtil.validateAndGetVersionComponents(api.getVersion(), api.getName());
+        List<API> apiVersions = apiConsumer.getAllAPIVersions(apiProvider, api.getName(), organization);
+        List<SemVersion> semVersionList = new ArrayList<>();
+        for (API apiVersion : apiVersions) {
+            if (apiVersion.getStatus().equals(APIConstants.PUBLISHED_STATUS)
+                    && apiVersion.getId().getVersion().startsWith("v")) {
+                try {
+                    SemVersion currentSemVersion = SemanticVersionUtil.validateAndGetVersionComponents(
+                            apiVersion.getId().getVersion(), api.getName());
+                    if (currentSemVersion.getMajor() == semVersion.getMajor()) {
+                        semVersionList.add(currentSemVersion);
+                    }
+                } catch (APIVersionValidationFailureException e) {
+                    log.warn("Ignored API version: " + apiVersion.getId().getVersion() +
+                            " when calculating latest major version of API: " + api.getUuid());
+                }
+            }
+        }
+        Collections.sort(semVersionList);
+        SemVersion latestVersionInMajorRange = semVersionList.get(0);
+
+        MajorRangeVersionInfo majorRangeVersionInfo;
+        if (latestVersionInMajorRange.getVersion().equals(api.getVersion())) {
+            majorRangeVersionInfo = new MajorRangeVersionInfo(true, api.getVersion(), api.getUuid());
+        } else {
+            API latestMajorVersionAPI = apiConsumer.getAPIInfoByNameVersion(
+                    apiProvider, api.getName(), latestVersionInMajorRange.getVersion(), organization);
+            majorRangeVersionInfo = new MajorRangeVersionInfo(false,
+                    latestMajorVersionAPI.getId().getVersion(), latestMajorVersionAPI.getUuid());
+        }
+        VersionInfo versionInfo = new VersionInfo(api.getVersion(), majorRangeVersionInfo);
+        api.getApi().setVersionInfo(versionInfo);
+    }
+
     private APIDTO getAPIByAPIId(String apiId, String organization) throws APIManagementException {
         try {
             APIConsumer apiConsumer = RestApiCommonUtil.getLoggedInUserConsumer();
             ApiTypeWrapper api = apiConsumer.getAPIorAPIProductByUUID(apiId, organization);
             String status = api.getStatus();
+            // Try adding major version range latest version details to API's additional properties
```

Comment:
We are not adding to additional properties.

Reply:
Fixed with: https://github.com/wso2/carbon-apimgt/pull/12107/commits/76e6b2935490b38e1aa847a99a59d790015b8495